mesos/cli/cmds/frameworks.py

```diff
@@ -0,0 +1,144 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+
+from __future__ import absolute_import, print_function
+
+import blessings
+import prettytable
+
+from .. import cli
+from ..master import CURRENT as MASTER
+
+try:
+    from collections import OrderedDict
+except ImportError:
+    from ordereddict import OrderedDict
+
+
+parser = cli.parser(
+    description="frameworks list"
+)
+
+parser.add_argument(
+    "-t", "--task-breakdown", action="store_true",
+    help="show task status breakdown"
+)
+
+parser.add_argument(
+    "-i", "--inactive", action="store_true",
+    help="show inactive frameworks as well"
+)
+
+
+class TaskStats:
```

Comment:
As a rule, memoization is used extensively throughout the mesos-cli codebase. This should be baked directly into the native objects themselves.